docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+
+[[docs-replication]]
+== Reading and Writing documents
+
+=== Introduction
+
+Each index in Elasticsearch is <<getting-started-shards-and-replicas,divided into shards>>
+and each shard can have multiple copies. These copies are known as `replication group` and must be kept in sync when documents
+are added to and removed. If we fail to do so, reading from one copy will result in very different results than reading from another.
+The process of keeping the shard copies in sync and serving reads from them is what we call the *data replication model*.
+
+Elasticsearch’s data replication model is based on the primary-backup model and is described very well in the https://www.microsoft.com/en-us/research/publication/pacifica-replication-in-log-based-distributed-storage-systems/[PacificA paper] of
+Microsoft Research. That model is based on having a single copy from the replication group that is acting as the primary shard.
+The other copies are called “replica shards”. The primary serves as the main entry point for all indexing operations. It is in charge of
+validating them and making sure they are correct. Once an index operation has been accepted by the primary, the primary is also
+responsible for replicating the operation to the other copies.
+
+This purpose of this section is to give a high level overview of the Elasticsearch replication model and discuss the implications
+it has for various interactions between write and read operations.
+
+=== Basic write model
+
+Every indexing operation in Elasticsearch is first resolved to a replication group.
```

Comment:
to a replication group using <<index-routing,routing>>, typically based on the document ID.